teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -50,6 +52,10 @@ interface VcsRootLocator {
 interface BuildLocator {
     fun fromConfiguration(buildConfigurationId: BuildConfigurationId): BuildLocator
 
+    fun withNumber(buildNumber: String): BuildLocator
+
+    fun withVcsRevision(vcsRevision: String): BuildLocator
```

Comment:
The name looks strange, what does it mean 'build with revision'? I would rename it to something like `containingVcsRevision`.
Also it's not evident will it return all the builds containing the specified change, or only the first one.

Reply:
Too many builds containingVcsRevision. It's about builds built on exact vcsRevision. I'll add a javadoc. But the name "withVcsRevision" is correct

Reply:
Maybe name it `fromVcsRevision` then?

Reply:
`fromVcsRevision` implies that further revisions will be included too, so I'm for retaining `withVcsRevision` name